### ISSUE TITLE
Type check literal kwargs from unpacked dictionaries

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#10582]
Consider the snippet below - currently it does fail on runtime, but does pass type checking.

```python
# pyright: strict

def test(a: int, b: int): ...

# Static: no type errors:
# Runtime: TypeError: test() got an unexpected keyword argument 'c'
test(**{"a": 25, "b": 42, "c": 11})
```

Personally, I would rather convert it to `test(a=25, b=42, c=11)` to make it more idiomatic (and a little micro bit more performant), but I see users sometimes do use dicts + unpacking to pass it as kwargs. Though it's not clear why, if function has no `**kwargs` and unpacked dictionary doesn't have any dynamic items.

In theory, it would be better to have a linter rule for this, but currently linters seem to operate with very little type information (though in the future maybe Ruff + ty might be able to provide more advanced linter rules) and they won't be able to detect it precisely enough.

**Comment by erictraut** [comment]
It's unlikely that we'd implement such a feature. It's not typical to use a literal dictionary and then unpack it directly when calling a function. Plus, such a change would introduce inconsistencies in the way that pyright evaluates types and would require a bunch of hard-to-maintain special casing.

If you want better type checking for this code, you should use keyword arguments when making the call.
